lib/ReactViews/Tools/DeltaTool/DeltaTool.jsx

```diff
@@ -0,0 +1,221 @@
+import dateFormat from "dateformat";
+import PropTypes from "prop-types";
+import React, { useEffect, useLayoutEffect, useState } from "react";
+import createCatalogMemberFromType from "../../../Models/createCatalogMemberFromType";
+import raiseErrorToUser from "../../../Models/raiseErrorToUser";
+import Styles from "./delta-tool.scss";
+import DatePickers from "./DatePickers";
+import LocationPicker from "./LocationPicker";
+import prettifyCoordinates from "../../../Map/prettifyCoordinates";
+
+/**
+ * A tool for comparing imagery at two points in time.
+ */
+function DeltaTool({ terria, tool, onCloseTool }) {
+  const { type, item: catalogItem } = tool;
+  if (
+    type !== "delta" ||
+    !(catalogItem && catalogItem.supportsDeltaComparison)
+  ) {
+    return null;
+  }
+
+  useEffect(() => {
+    // On mount; to avoid showing both imageries at once, we temporarily disable the parent catalog item.
+    catalogItem.isEnabled = false;
+    return () => {
+      // Re-enable on unmount
+      catalogItem.isEnabled = true;
+    };
+  }, []);
+
+  const [item, setItem] = useState(undefined);
+  const [location, setLocation] = useState(undefined);
+  const [primaryDate, setPrimaryDate] = useState(catalogItem.discreteTime);
+  const [secondaryDate, setSecondaryDate] = useState(catalogItem.discreteTime);
+  const [pickerMessage, setPickerMessage] = useState(
+    "Select a point by clicking on the map"
+  );
+
+  // Duplicate the catalog item
+  useLayoutEffect(() => {
```

Comment:
Does this need to happen before paint? (e.g. can it be moved into the earlier `useEffect`?) I'd rather use placeholder things and _then_ do any duplication required, than block things with this.